core/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByScrollRequest.java

```diff
@@ -152,8 +156,8 @@ public ActionRequestValidationException validate() {
                             + size + "]",
                     e);
         }
-        if (searchRequest.source().slice() != null && slices != 1) {
-            e = addValidationError("can't specify both slice and workers", e);
+        if (searchRequest.source().slice() != null && slices != DEFAULT_SLICES) {
+            e = addValidationError("can't set a specific single slice for this request and multiple slices", e);
```

Comment:
Yeah I think that makes sense. When I wrote that I was thinking about the child tasks having a slice builder set in automatic slicing, but this aligns more with how the api is described